Wikipedia/UI-V5/WMFHomeViewController.m

```diff
@@ -427,6 +435,9 @@ - (void)controller:(id<WMFHomeSectionController>)controller didSetItems:(NSArray
 
 - (void)controller:(id<WMFHomeSectionController>)controller didAppendItems:(NSArray*)items {
     NSInteger section = [self indexForSectionController:controller];
+    if (section == NSNotFound) {
```

Comment:
actually, can't we just set the controller's `delegate` to `nil` and prevent callbacks after it has been unloaded?

also, do you think now would be an OK time to start documenting this class? either that, or make the implementation a bit more expressive so the structure is clearer? hard to tell what should/shouldn't be happening in the various methods and what the guarantees are